expand single darkroom module: make parameter work across groups

elstoc opened the pull request:
ensures that, if the "expand single darkroom module" preference is set, this setting is observed across all module groups.

I've also raised #5129 for discussion about whether a different (better?) solution might be wanted longer term. For now, this fix makes sure the 'expand single module' parameter always works consistently even when changing group.

TurboGit commented:
I don't see why this should be better. The idea was to have a single opened module at the same time visible. So having one per module group is fine especially since you can switch from one group to another to adjust two modules without having to reopen them all the time. I'm used to that.

elstoc commented:
I wouldn't mind if it worked that way but it doesn't, because of the 'all modules', 'active modules' and 'favourites' groups. See the more detailed discussion in #5129.

elstoc commented:
I think the only way around this might be a preference. I do see your use case, but the main problem is that while I don't mind one module open per group, if I go to favourites or active modules and find three modules expanded because they're each expanded in one of the other groups, the click-move-click to get back a nice state is painful.

TurboGit commented:
Probably OBE by #5683 ?

elstoc commented:
> Probably OBE by #5683 ?

I don't think it is. This will continue to be an issue if a module can be present in more than one group unless there's a fix in #5683 that I didn't notice.

TurboGit commented:
Ok, will see... Thanks for the quick feedback!

elstoc commented:
I've rebased and retested. The issue still occurs so, as discussed, I've added a new preference setting to allow the former behaviour to be retained if desired.

TurboGit commented on data/darktableconfig.xml.in:
```diff
@@ -1536,6 +1536,13 @@
     <shortdescription>expand a single darkroom module at a time</shortdescription>
     <longdescription>this option toggles the behavior of shift clicking in darkroom mode</longdescription>
   </dtconfig>
+  <dtconfig prefs="darkroom">
```
As this option require another option to be set I would propose a combo box with multiple choices:

name :
**control expanded modules**

choices:

- no
- single visible module
- single module

With a tooltip with more explanations and a XML snippet for the documentation.

elstoc replied:
Ok good idea

elstoc replied:
Actually now I think about it I've remembered why I didn't do this. The problem is that the current darkroom/ui/single_module option can have its logic reversed by shift+clicking on a module header and that can't be easily achieved with a 3-way combobox. If you selected 'no' in the above combobox how should shift-click behave?

TurboGit replied:
I see! I didn't even know about the Shift-Click behavior :) So the PR is ok.

TurboGit reviewed (approved):
Thanks !

TurboGit commented:
@elstoc : can you propose an entry for the doc ?

elstoc commented:
ok
